### DISCUSSION TITLE
iOS build failed!

### DISCUSSION THREAD

**Discussion by sorydima** [codemagic-ci-cd/codemagic-docs#2538] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Try to build https://github.com/sorydima/REChain-.git

### Expected results

Build done ..

### Actual results

You are in 'detached HEAD' state. You can look around, make experimental
         changes and commit them, and you can discard any commits you make in this
         state without impacting any branches by switching back to a branch.

         If you want to create a new branch to retain commits you create, you may
         do so (now or later) by using -c with the switch command. Example:

           git switch -c <new-branch-name>

         Or undo this operation with:

           git switch -

         Turn off this advice by setting config variable advice.detachedHead to false

       configure: WARNING: Can't find Tcl configuration definitions
       configure: WARNING: *** Without Tcl the regression tests cannot be executed ***
       configure: WARNING: *** Consider using --with-tcl=... to define location of Tcl ***
         Cloning into '/var/folders/m7/h1mg7c7x40ddjz6mxjxm3htr0000gn/T/d20240517-3129-1et1fy'...
         Note: switching to 'db0c122b671bc9760385e0355be00eede3b7bb44'.

         You are in 'detached HEAD' state. You can look around, make experimental
         changes and commit them, and you can discard any commits you make in this
         state without impacting any branches by switching back to a branch.

         If you want to create a new branch to retain commits you create, you may
         do so (now or later) by using -c with the switch command. Example:

           git switch -c <new-branch-name>

         Or undo this operation with:

           git switch -

         Turn off this advice by setting config variable advice.detachedHead to false

         % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                        Dload  Upload   Total   Spent    Left  Speed
       
     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       
   100 36.0M  100 36.0M    0     0  66.6M      0 --:--:-- --:--:-- --:--:-- 66.6M

Error running pod install

Build failed :|
Failed to build for iOS

![Снимок экрана от 2024-05-17 18-56-49](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/691e450b-4883-4c0c-aa19-121324d689ae)
![Снимок экрана от 2024-05-17 18-56-54](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/eed77ecd-54c7-40e7-bf0b-70594dc909a8)
![Снимок экрана от 2024-05-17 18-57-05](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/5caf734b-3eb9-421a-83bf-32cd6386be9f)

### Build id (optional)

66477cbb8657e78e41b67dc2

**Comment by dtrdic** [comment]
Hello @sorydima I can see the following error prior to the "detach head" error:
`[!] The version of CocoaPods used to generate the lockfile (1.12.1) is higher than the version of the current executable (1.11.3)`
As the error message suggests, your Podfile.lock file was generated with a different CocoaPods version, which may result in this incompatibility issue. Try removing your Podfile.lock file from your project, and clean your project by running these commands locally:
```
cd ios
pod cache clean --all
rm -rf ~/Library/Caches/CocoaPods
rm -rf Pods
rm -rf ~/Library/Developer/Xcode/DerivedData/*
pod deintegrate
flutter clean
pod install
```

**Reply by sorydima** [reply]
Yep, I understand how to do the ready build on local machine but nowadays I don't have mac)) So I use CodeMagic to build the app for iOS and mac OS. But it's still an error for the both ones...

**Reply by dtrdic** [reply, marked as answer]
@sorydima have you tried to specify different Cocoapods version in Workflow editor? I can see you used `Cocoapods 1.11.3 `version, try change it to `1.12.1` or use the default version
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/c4096863-db8a-4b75-bc79-940c4e313885)

**Comment by sorydima** [comment]
Guys, I don't have a mac computer
